### PR TITLE
Feature/design update

### PR DESCRIPTION
**Pull request by SkylerPIlot**
Organized Robotmap IDs and Created Getters for all sensor values.

### CODE REVIEW

File: src/RobotMap.h [hunk 1]
```diff
@@ -6,29 +6,50 @@
...
+
+  //..........GearIntake..........
+  static const int kIDGearIntake = 10;
+
+  //..........HopperIntake..........
+  static const int kIDHopperIntake = 9;
+
+  //..........Lifter..........
+  static const int kIDLifterLeft = 11;
+  static const int kIDLifterRight = 12;
```
Comment (OliviaRS) [inline src/RobotMap.h]: Do not use the name of a subsystem when naming a talon

File: src/RobotMap.h [hunk 2]
```diff
@@ -6,29 +6,50 @@
...
+
+  //..........Shooter..........
+  static const int kIDFlyWheelTop1 = 5;
+  static const int kIDFlyWheelTop2 = 6;
+  static const int kIDFlyWheelBottom = 7;
```
Comment (OliviaRS) [inline src/RobotMap.h]: Inconsistencies on whether "where" or "what" comes first. Pick one.

File: src/RobotMap.h [hunk 3]
```diff
@@ -6,29 +6,50 @@
...
+
+//..........Pneumatics..........
+
+  //..........GearPneumatics..........
   static const int kIDpickupForward = 1;
   static const int kIDpickupReverse = 6;
 
+  //..........ShiftingPneumatics..........
+  static const int kIDShitftingHigh = 0;
+  static const int kIDShitftingLow = 7;
```
Comment (OliviaRS) [inline src/RobotMap.h]: Should still be Forward and Reverse. High and Low belong in the subsystem.

File: src/Subsystems/BallConveyor.cpp [hunk 4]
```diff
@@ -15,26 +15,31 @@ std::shared_ptr<BallConveyor> BallConveyor::getInstance() {
...
 	// Set the default command for a subsystem here.
 	// SetDefaultCommand(new MySpecialCommand());
   SetDefaultCommand(new ::ConveyorControl(0.0, 0.0));
 }
-void BallConveyor::ConveyorControl(double conveyorSpeed,double moverSpeed){
+void BallConveyor::ConveyorControl(double conveyorSpeed){
 
+  ConveyorMotor.Set(conveyorSpeed);
+
+}
```
Comment (OliviaRS) [inline src/Subsystems/BallConveyor.cpp:31]: The next method is a good example of how much white space there should be. This is too much.

File: src/Subsystems/Chassis.h [hunk 5]
```diff
@@ -13,15 +14,18 @@ class Chassis: public frc::Subsystem {
 
   CANTalon right1Wheel;
 	CANTalon right2Wheel;
-	CANTalon right3Wheel;
 	CANTalon left1Wheel;
 	CANTalon left2Wheel;
-	CANTalon left3Wheel;
+	DoubleSolenoid (Shifter);
```
Comment (OliviaRS) [inline src/Subsystems/Chassis.h]: Unnecessary parentheses

File: src/Subsystems/GearPickup.h [hunk 6]
```diff
@@ -11,9 +12,9 @@ class GearPickup : public Subsystem {
   static const char kSubsystemName[];
   static std::shared_ptr<GearPickup> self;
 
+  CANTalon GearRoller;
   DoubleSolenoid (Pickup);
```
Comment (OliviaRS) [inline src/Subsystems/GearPickup.h:16]: Unnecessary parentheses

File: src/Subsystems/Intake.cpp [hunk 7]
```diff
@@ -28,16 +30,13 @@ void Intake::InitDefaultCommand() {
 }
 
 void Intake::SetIntake(double speed){
-  rightMotor.Set(speed);
-  leftMotor.Set(speed);
+
+  HopperRoller.Set(speed);
 
 }
```
Comment (OliviaRS) [inline src/Subsystems/Intake.cpp:31]: Braces go on the next line, not the line after that.

File: src/Subsystems/Intake.h [hunk 8]
```diff
@@ -11,8 +11,7 @@ class Intake : public frc::Subsystem {
 	static const char kSubsystemName[];
 	static std::shared_ptr<Intake> self;
 
-	CANTalon rightMotor;
-	CANTalon leftMotor;
+	CANTalon HopperRoller;
```
Comment (OliviaRS) [inline src/Subsystems/Intake.h]: Keep your casing consistent

File: src/Subsystems/Intake.h [hunk 9]
```diff
@@ -22,10 +21,7 @@ class Intake : public frc::Subsystem {
 
 	void SetIntake(double speed);
 
-	double GetIntake();
-
-
-
+	double GetIntakeRPM() const;
 
 };
```
Comment (OliviaRS) [inline src/Subsystems/Intake.h:25]: Whitespace

File: src/Subsystems/Shooter.cpp [hunk 10]
```diff
@@ -31,16 +39,27 @@ void Shooter::InitDefaultCommand() {
...
+void Shooter::SetShooter(double RPM){
+
+  FlyWheelTop1.Set(RPM);
+  FlyWheelBottom.Set(RPM * .75);
+
+}
 
-  rightMotor.Set(speed);
-  leftMotor.Set(speed);
+bool Shooter::IsLeftTurretAlligned() const{
```
Comment (OliviaRS) [inline src/Subsystems/Shooter.cpp]: Hallefect sensors go on chassis

**Review by OliviaRS — APPROVED** [review]

File: src/Subsystems/Turret.h [hunk 11]
```diff
@@ -23,6 +23,10 @@ class Turret : public frc::Subsystem {
 	void SetTurret(double speed);
 
 	double GetTurret();
+
+	bool IsLeftSwitchPressed() const;
+//TODO Right Make sure these values correlate with how they go on the robot
+	bool IsRightSwitchPressed() const;
```
Comment (OliviaRS) [inline src/Subsystems/Turret.h:27]: Make it clear that this means the turret has hit the stopper by name

**Review by OliviaRS — CHANGES_REQUESTED** [review]

**Comment by duane701** [comment]
Replaced by #20

**Pull request closed without merging**